### PR TITLE
feat: CS & SS configurability against host header injection

### PR DESCRIPTION
**Pull request by andresrosenthal**
Refs: XRDDEV-2387

### CODE REVIEW

File: src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/config/AllowedHostsConfig.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityCustomizer;
+import org.springframework.security.web.firewall.StrictHttpFirewall;
+import org.springframework.util.CollectionUtils;
+
+@Configuration
+@ComponentScan(basePackages = {"org.niis.xroad.cs.admin"})
```
Comment (ricardas-buc) [inline src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/config/AllowedHostsConfig.java:38]: do we need extra component scan here?

File: src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/config/AllowedHostsConfig.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+public class AllowedHostsConfig {
+
+    @Bean
+    public WebSecurityCustomizer webSecurityCustomizer(AdminServiceProperties properties) {
```
Comment (ricardas-buc) [inline src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/config/AllowedHostsConfig.java:42]: both cs and ss beans seem identical, why not move to common-rest-api? Props can be passed similarly to ip throttling or api client settings.

File: src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/config/AllowedHostsConfig.java (hunk 1 above)
Comment (andresrosenthal) [reply src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/config/AllowedHostsConfig.java:38]: Probably not, due to IDE complaint didn't think twice

File: src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/config/AllowedHostsConfig.java (hunk 2 above)
Comment (andresrosenthal) [reply src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/config/AllowedHostsConfig.java:42]: Indeed, good catch

**Pull request merged**